SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorPackageAssembly.cpp

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKEditorPackageAssembly.h"
+
+#include "Async/Async.h"
+#include "Framework/Notifications/NotificationManager.h"
+#include "Misc/App.h"
+#include "Misc/FileHelper.h"
+#include "UnrealEdMisc.h"
+
+#include "SpatialGDKEditorModule.h"
+#include "SpatialGDKEditorSettings.h"
+#include "SpatialGDKServicesConstants.h"
+#include "SpatialGDKServicesModule.h"
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKEditorPackageAssembly);
+
+#define LOCTEXT_NAMESPACE "SpatialGDKEditorPackageAssembly"
+
+namespace
+{
+	const FString SpatialBuildExe = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory(TEXT("SpatialGDK/Binaries/ThirdParty/Improbable/Programs/Build.exe"));
+	const FString Linux = TEXT("Linux");
+	const FString Win64 = TEXT("Win64");
+}
+
+FSpatialGDKPackageAssembly::FSpatialGDKPackageAssembly()
+{
+}
+
+static FString GetStagingDir()
+{
+	return SpatialGDKServicesConstants::SpatialOSDirectory / TEXT("build/unreal");
+}
+
+void FSpatialGDKPackageAssembly::BuildAssembly(const FString& ProjectName, const FString& Platform, const FString& Configuration, const FString& AdditionalArgs)
+{
+	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
+	FString WorkingDir = FPaths::ConvertRelativePathToFull(FPaths::ProjectDir());
+	FString Project = FPaths::ConvertRelativePathToFull(FPaths::GetProjectFilePath());
+	FString Args = FString::Printf(TEXT("%s %s %s %s %s"), *ProjectName, *Platform, *Configuration, *Project, *AdditionalArgs);
+	PackageAssemblyTask = MakeShareable(new FMonitoredProcess(SpatialBuildExe, Args, WorkingDir, true));
```

Comment:
line 43 - 47 and line 65 - 69 seem to be identical. Might be worthwhile puling that out?